### PR TITLE
Port overlay

### PR DESCRIPTION
**Pull request by inderpreetsingh**
https://youtu.be/LZeIU_xJrzo

Now the search for ports happen when you open window and click open the toggle.

### CODE REVIEW

File: www/js/overlay.js [hunk 1]
```diff
@@ -13,8 +13,12 @@ const { store } = remote.require('./app');
 const analytics = remote.getGlobal('analytics');
 
 const { overlayVars } = store.get('obs').overlayPrefs;
+let overlayCast = store.getUserPref('app.overlay-cast');
 
-const url = `http://${host}:${overlayPort}/`;
+const controlPanel = document.querySelector('.control-panel');
+const webview = document.createElement('webview');
+
+let url = `http://${host}:${overlayPort}/`;
```
Comment (tsingh777) [inline www/js/overlay.js]: copy button's url doesn't get updated when initially off

File: app.js [hunk 2]
```diff
@@ -301,6 +282,41 @@ const shouldQuit = app.makeSingleInstance(() => {
...
+      http.listen(port);
+    },
+  );
+};
+
+ipcMain.on('toggle-obs-cast', (event, arg) => {
+  if (arg) {
+    searchPorts();
+  } else {
+    http.close();
```
Comment (tsingh777) [inline app.js]: Doesn't look overlay actually turns off. updating lines still updates the webpage.

File: www/js/overlay.js [hunk 3]
```diff
@@ -309,11 +339,12 @@ const themeSwatchFactory = themeOptions => {
 
 themeSelector.appendChild(h('div.theme-selector-header', 'Presets'));
 
+document.body.classList.toggle('overlay-off', !overlayCast);
```
Comment (tsingh777) [inline www/js/overlay.js]: the buttons still work when toggle is off

File: app.js (hunk 2 above)
Comment (inderpreetsingh) [reply app.js]: Yes, I have shared about that in `sttm-desktop-dev`, please take a look. 🙏 That's the default behaviour of http.close(). It won't kill existing connections but prevent new connections.

File: www/js/overlay.js (hunk 1 above)
Comment (inderpreetsingh) [reply www/js/overlay.js]: Yes, but it gets updated as soon as we click the on/off switch.

File: www/js/overlay.js (hunk 3 above)
Comment (inderpreetsingh) [reply www/js/overlay.js]: Should be fixed now.

File: www/src/scss/_overlay.scss [hunk 4]
```diff
@@ -157,6 +157,52 @@ input[type='color'] {
   width: 60px;
 }
 
+.overlay-off {
+  .cp-icon,
+  .setting-label {
+    color: #ddd;
+    cursor: not-allowed;
+    pointer-events: none;
```
Comment (tsingh777) [inline www/src/scss/_overlay.scss]: minor thing: larivaar/padched button and copy url button still seem to work when disabled.

File: www/js/overlay.js [hunk 5]
```diff
@@ -195,6 +201,31 @@ const toggleLarivaar = h(
   h('div.setting-label', `Use ${overlayVars.overlayLarivaar ? 'Padched' : 'Larivaar'}`),
 );
 
+const toggleCast = h(
+  'div.input-wrap.cast-toggle',
+  {
+    onclick: evt => {
```
Comment (tsingh777) [inline www/js/overlay.js]: perhaps we should also send empty slide if toggle is being turned off? doesn't have to be this PR.

**Pull request merged**